rust/src/tests/errors/etxtbsy.rs

```diff
@@ -0,0 +1,68 @@
+/// Create a test case which asserts that it returns ETXTBSY
+/// when the file is a pure procedure (shared text) file that is being executed.
+/// There are multiple forms for this macro:
+///
+/// - A basic form which takes the syscall, and optionally a `~path` argument
+///   to indicate where the `path` argument should be substituted if the path
+///   is not the only argument taken by the syscall.
+///
+/// ```
+/// // `unlink` accepts only a path as argument.
+/// enotdir_comp_test_case!(unlink);
+/// // `chflags` takes a path and the flags to set as arguments.
+/// // We need to add `~path` where the path argument should normally be taken.
+/// enotdir_comp_test_case!(chflags(~path, FileFlags::empty()));
+/// ```
+///
+/// - A more complex form which takes multiple functions
+///   with the context and the path as arguments for syscalls
+///   requring to compute other arguments.
+///
+/// ```
+/// enotdir_comp_test_case!(chown, |ctx: &mut TestContext, path: &Path| {
+///   let user = ctx.get_new_user();
+///   chown(path, Some(user.uid), None)
+/// })
+/// ```
+macro_rules! etxtbsy_test_case {
+    ($syscall: ident, $($f: expr),+) => {
+        crate::test_case! {
+            #[doc = concat!(stringify!($syscall),
+                 " returns ETXTBSY when the file is a pure procedure (shared text) file that is being executed.")]
+            etxtbsy
```

Comment:
There should also be a guard asserting that the file system is mounted executably.

Reply:
Should it check that the file system is mounted with the `noexec` flag? Because in that case, this would require to check the `f_flags` field of [`Statfs`](https://docs.rs/nix/latest/nix/sys/statfs/struct.Statfs.html) which isn't returned by Nix. It is returned by [`Statvfs`](https://docs.rs/nix/latest/nix/sys/statvfs/struct.Statvfs.html), however it doesn't support the `noexec` flag on FreeBSD.

Reply:
Looks like we need to extend Nix.  I'll do it.

Reply:
Ok, here it is.  With this change, you can get the mount flags using `statfs`.
https://github.com/nix-rust/nix/pull/1849

Reply:
Thanks for implementing it!